python/custom_test_files.py

```diff
@@ -0,0 +1,189 @@
+            ts_row = f.readline() #Reading in the first row as the TS submission. 
+            lar_rows = f.readlines() #Reading in the other rows as the LAR submission.  
+        with open("../edits_files/file_parts/ts_data.txt", 'w') as out_ts: #Writes a file of TS data. 
+            out_ts.writelines(ts_row)
+            
+        statement = """ TS file saved as: "../edits_files/file_parts/lar_data.txt" """
+        ts_filepath = "../edits_files/file_parts/ts_data.txt"
+        return ts_filepath #returns a string of the TS filepath.
+    
+    def separate_lar(self):
+        with open(self.filename, 'r+' ) as f: #Opening the submission file. 
+            ts_row = f.readline() #Reading in the first row as the TS submission. 
+            lar_rows = f.readlines() #Reading in the other rows as the LAR submission.  
+        with open("../edits_files/file_parts/lar_data.txt", 'w') as out_lar: #Writes a file of LAR data. 
+            out_lar.writelines(lar_rows)
+        
+        #Prints the location of each data file. 
+        statement = """LAR file saved as: "../edits_files/file_parts/lar_data.txt" """
+        
+        lar_filepath = "../edits_files/file_parts/lar_data.txt"
+        return lar_filepath #returns a string of the LAR filepath.
+    
+    def get_rows(self, lar_file=None, row_count=None):
+        """
+        Returns a data frame of LAR data with a specified number of rows.
+        
+        """
+        with open('../schemas/lar_schema.json') as d: #Loads in the schema for the LAR file as a dataframe. 
+            headers = json.load(d)
+            headers = pd.DataFrame(headers)
+        #Reads in the data from lar_file. 
+        df = pd.read_csv(lar_file, delimiter="|", header=None, dtype='object',
+                         keep_default_na=False) 
+        
+        df.columns = headers['field'] #Takes the field names from the LAR schema as column names.
+        
+        current_row = len(df.index) #Number of rows currently in the dataframe. 
+
+        #Calculates a multiplier taking the ceiling function of desired row count over current row count. 
+        multiplier = math.ceil(row_count/current_row)
+        
+        #Concatenates data to produce the number of rows by the multiplier in a new LAR dataframe. 
+        df = pd.concat([df]*int(multiplier))
+        
+        #Drops the number of rows to the row count specified. 
+        
+        if (row_count%current_row) == 0:
+            return df
+        else:
+            drop_rows = current_row - (row_count%current_row)
+            df = df[:-(drop_rows)]
+            return df
+         
+    def assign_uli(self, df=None, ts_file=None):
+        """
+        Assigns new ULI's to a dataframe of lar rows, based on the bank number in the TS file. 
+        
+        """
+        #Reads in ts_file.
+        ts_df = pd.read_csv(ts_file, delimiter="|", header=None, dtype='object',
```

Comment:
If you modify the separate_ts function to create a self.ts_row then you can pull the data you need from memory instead of having to read a file, that will help you with a bit of efficiency.